packages/app-desktop/services/sortOrder/PerFolderSortOrderService.ts

```diff
@@ -126,6 +126,8 @@ export default class PerFolderSortOrderService {
 	private static getSelectedFolderId(): string {
 		if (this.folderState.notesParentType === 'Folder') {
 			return this.folderState.selectedFolderId;
+		} else if (this.folderState.notesParentType === 'SmartFilter') {
+			return 'all_notes_folder_id';
```

Comment:
I don't understand this logic. Does it mean that the "All notes" item is defined somewhere as a smart filter? But then I believe it also has a special ID (which you use in a previous of the code), yet here you give it yet another ID.

Also it's obvious that this is going to break if there's ever more than one smart filter since they will all be detected as the "All notes" item. For example the trash folder is a smart filter so maybe the logic is already broken.

Reply:
Right, let me explain concisely where that's coming from:

In `PerToggleSortOrderService.ts`, the `eventManager` is listening to two variables:
```typescript
eventManager.appStateOn('notesParentType', this.onFolderSelectionMayChange.bind(this, 'notesParentType'));
eventManager.appStateOn('selectedFolderId', this.onFolderSelectionMayChange.bind(this, 'selectedFolderId'));
```
In `reducer.ts`, you have this `export type NotesParentType = 'Folder' | 'Tag' | 'SmartFilter';`. As you said, the `All Notes` is defined as a `SmartFilter` whenever I click on that. I used the ID in the previous code, and now I use a different ID because I was not sure whether I could use the special ID, and I forgot to ask. 

If the new `Trash` folder or other folders in the future are also `SmartFilter` type, then this breaks.

Reply:
I think I have two solutions:

### Solution 1
We will add more types to the new `FolderType`. Then, we can quickly scale up the number of different folder types we need. (There is already an existing logic with `NotesParentType` and cannot change the notesParentType for the `All Notes` and `Trash`, so better not to change anything on that.)

Then, we could create a new variable in `PerToggleSortOrderService.ts` like `folderType`, let eventManager listen to it, and use it in the `getSelectedFolderId()`. We would have to change some functions in `reducer.ts`.

### Solution 2
We would only concentrate on the `togglePerFolderSortOrder` command, where we add one more parameter (folderType) to the `PerFolderSortOrderService.set(folderId, folderType, own)`. Then we would have:

```typescript
private static getSelectedFolderId(): string {
		if (this.folderState.notesParentType === 'Folder') {
			return this.folderState.selectedFolderId;
		} else if (this.folderState.notesParentType === 'SmartFilter') {
			if (this.folderType == 'All Notes') {
                             return "ALL NOTES ID"
                         } 
                         if (this.folderType == 'Trash') {
                             return "TRASH ID"
                         }
		} else {
			return '';
		}
	}
```
Instead of doing if statements, we could refactor them to switch statements for better readability.

Reply:
I don't quite follow all the details but the function says `getSelectedFolderId` and "All notes" is a smart filter that has an ID "ALL_NOTES_FILTER_ID". So why not just return the ID? Why doesn't just returning `this.folderState.selectedFolderId` work here? Or `selectedFolderId` is not set as expected?

Reply:
Right, it's because for `SmartFilter`, it's `selectedSmartFilterId` instead of `selectedFolderId`.

Reply:
Thanks for looking into it. Looks like you got it working without changing this PerFolderSortOrderService.ts in the end?

Reply:
Exactly, I just went with the previous solution of calling two functions in onClick.